tools/tesseratos/src/tesseratos/toolbox/plugin.cpp

```diff
@@ -0,0 +1,71 @@
+#include <imgui.h>
+
+#include <cubos/engine/imgui/plugin.hpp>
+
+#include <tesseratos/toolbox/plugin.hpp>
+
+using cubos::core::ecs::EventWriter;
+using cubos::core::ecs::Read;
+using cubos::core::ecs::Write;
+
+using cubos::core::ecs::Write;
+using cubos::engine::Cubos;
+
+using namespace tesseratos;
+
+bool Toolbox::isOpen(std::string toolName)
+{
+    if (mToolsMap.contains(toolName))
+    {
+        return mToolsMap[toolName];
+    }
+    mToolsMap[toolName] = false;
+    return false;
+}
+
+void Toolbox::open(std::string toolName)
+{
+    mToolsMap[toolName] = true;
+}
+
+void Toolbox::close(std::string toolName)
+{
+    mToolsMap[toolName] = false;
+}
+
+void Toolbox::doClick(std::string toolName)
+{
+    mToolsMap[toolName] = !mToolsMap[toolName];
+}
+
+auto Toolbox::begin()
+{
+    return mToolsMap.begin();
+}
+
+auto Toolbox::end()
+{
+    return mToolsMap.end();
+}
+
+static void showToolbox(Write<Toolbox> toolbox)
+{
+    ImGui::Begin("Toolbox");
+
+    for (auto it = toolbox->begin(); it != toolbox->end(); ++it)
```

Comment:
using the modern for loop would make this more readable. something like `auto& [tool, open] : toolbox`